ui/pages/setup_secure.tsx

```diff
@@ -1,19 +1,92 @@
-import { Button } from "@canonical/react-components";
+import { SettingsFlow, UpdateSettingsFlowBody } from "@ory/client";
 import type { NextPage } from "next";
+import { useRouter } from "next/router";
+import { useEffect, useState, useCallback } from "react";
 import React from "react";
+import { handleFlowError } from "../util/handleFlowError";
+import { Flow } from "../components/Flow";
+import { kratos } from "../api/kratos";
 import PageLayout from "../components/PageLayout";
+import { AxiosError } from "axios";
+import { Spinner } from "@canonical/react-components";
 
 const SetupSecure: NextPage = () => {
+  const [flow, setFlow] = useState<SettingsFlow>();
+
+  // Get ?flow=... from the URL
+  const router = useRouter();
+  const { return_to: returnTo, flow: flowId } = router.query;
+
+  useEffect(() => {
+    // If the router is not ready yet, or we already have a flow, do nothing.
+    if (!router.isReady || flow) {
+      return;
+    }
+
+    // If ?flow=... was in the URL, we fetch it
+    if (flowId) {
+      kratos
+        .getSettingsFlow({ id: String(flowId) })
+        .then((res) => setFlow(res.data))
+        .catch(handleFlowError(router, "settings", setFlow));
+      return;
+    }
+
+    // Otherwise we initialize it
+    kratos
+      .createBrowserSettingsFlow({
+        returnTo: returnTo ? String(returnTo) : undefined,
+      })
+      .then(({ data }) => {
+        if (data.request_url !== undefined) {
+          window.location.href =
+            "http://localhost:4455/ui/setup_secure" + "?flow=" + data.id;
```

Comment:
issue: host and port should be externalized piece of info
this could easily break

Reply:
You are right, this should be a relative redirect. Pushed a commit to fix it.